homework_3/solution.js

```diff
@@ -0,0 +1,56 @@
+// solution.js (called by test.js)
+// module.exports = function (equation) {
+//      return '<Your result here>';
+//  }
+
+// function equation (Result) {   
+ 
+    var colors = require('colors');
+    var Result;
+    var NextOperand;
+    var Operator;
+    var IndexStart;
+
+    IndexStart = 4; // assume min 2 operands and go directly to 4th arg to start calc
+
+    Result = Number(process.argv[2]);
+    for (var i=IndexStart; i < process.argv.length; i+=2) {
+        NextOperand = Number(process.argv[i]);
+        Operator = (process.argv[i-1]);
+        if (Operator === '+') {
+            Result = Result + NextOperand;
+        } else if (Operator === '-') {
+            Result = Result - NextOperand;
+        } else if (Operator === 'x') {
+            Result = Result * NextOperand;
+        } else if (Operator === '*') {
+            Result = Result * NextOperand;
+        } else if (Operator === '/') {
+            Result = Result / NextOperand;  
+        } else if (Operator === '^') {  
+            Result = Math.pow(Result, NextOperand);    
+        } else  {        
+            //return('Operator not supported: ', Operator);
```

Comment:
This could be a good chance to throw an error: `throw new Error('Operator not supported');`